### PR TITLE
Change the way we install contrib packages

### PR DESCRIPTION
**Pull request by fabpot**
When using Flex "extensively", installing contrib repos is a bit frustrating. If you opt-out (the default), you have a warning that the recipe won't be installed. The way to fix it is to 1/ remove the package 2/ enable contrib 3/ reinstall the package.

Instead, what about asking the question to the user and act accordingly (that means less to learn as enabling contrib is done automatically when selecting `[p]`)?

### CODE REVIEW

**Comment by weaverryan** [comment]
I like this approach. And we would still have several months to try it out. This is the first time that Flex is interactive, correct? That's the only weird part... but I still think it's best.

File: src/Flex.php [hunk 1]
```diff
@@ -161,11 +160,44 @@ public function update(Event $event): void
...
         $this->io->writeError(sprintf('<info>Symfony operations: %d recipe%s</>', count($recipes), count($recipes) > 1 ? 's' : ''));
+        $installContribs = $this->composer->getPackage()->getExtra()['symfony']['allow-contrib'] ?? false;
         foreach ($recipes as $recipe) {
-            if ($recipe->isNotInstallable() && $recipe->isEmpty()) {
-                $this->io->writeError(sprintf('  - <warning>Ignoring</> recipe %s', $this->formatOrigin($recipe->getOrigin())));
-                $this->io->writeError('    Enable via composer config extra.symfony.allow-contrib true');
-                continue;
+            if ('install' === $recipe->getJob() && !$installContribs && $recipe->isContrib()) {
+                $this->io->writeError(sprintf('  - <warning>WARNING</> %s', $this->formatOrigin($recipe->getOrigin())));
+                $question = '    This package wants to update your project for better integration.
```
Comment (weaverryan) [inline src/Flex.php]: This doesn't make it look like it's clear that the issue is that this comes from the contrib repository. It's tough, we want to explain what's going on ("there are 2 repos, one is super protected, the other is not as protected, and this comes from that repo")... without all those words :).

Comment (fabpot) [reply src/Flex.php]: Indeed, the current sentence is far from perfect... I was expecting some help from you :)

File: src/Flex.php [hunk 2]
```diff
@@ -161,11 +160,44 @@ public function update(Event $event): void
...
+    Allow?
+    [<comment>y</>] Yes
+    [<comment>n</>] No
+    [<comment>a</>] Yes for all packages, only for the current installation session
+    [<comment>p</>] Yes permanently, never ask again for this project
+    (defaults to <comment>n</>): ';
```
Comment (xabbuh) [inline src/Flex.php]: Should we print a different text if the command is run in non-interactive mode?

File: src/Flex.php (hunk 1 above)
Comment (weaverryan) [reply src/Flex.php]: Indeed :). I don't remember if I was short on time... or just being lazy... Let's just pretend it was the first

>    The recipe for this package comes from the contrib repository, which is open to community contributions.
>    Allow this recipe?

Later, we could even add a short link to some docs (e.g. `http://symfony.com/contrib-recipes`)

Comment (fabpot) [reply src/Flex.php]: I've used:

```
The recipe for this package comes from the "contrib" repository, which is open to community contributions.
    Do you want to execute this recipe?
```

File: src/Flex.php (hunk 2 above)
Comment (fabpot) [reply src/Flex.php]: The question is never displayed if the command is non-interactive.

**Comment by fabpot** [comment]
This PR is ready

File: src/Flex.php (hunk 2 above)
Comment (fabpot) [reply src/Flex.php]: I've made one small tweak. If non-interactive mode, it displays 'IGNORING' like before this PR and 'WARNING' in interactive mode. So, in non-interactive mode, the display is the same as before.

**Pull request merged**